src/sentry/auth/helper.py

```diff
@@ -239,22 +239,25 @@ def handle_attach_identity(
         self,
         identity: Identity,
         member: Optional[OrganizationMember] = None,
+        user: Optional[User] = None,
     ) -> AuthIdentity:
         """
         Given an already authenticated user, attach or re-attach an identity.
         """
+        user = user or self.user
```

Comment:
should we replace the comment above? since we are potentially using a user passed in from `_get_user`, the user is not necessarily "authenticated" in the traditional sense (they haven't completed the login flow). Maybe "Given an authenticated user, or a user who has confirmed their email, attach or re-attach an identity"?)

Reply:
i'm wondering if it would be possible to clean up this a bit if we initialized the helper with a user, either request.user, `_get_user`, or `None`?

Reply:
Yeah, passing the user is a code smell. My first (bad) idea was to group the two affected methods into a nested helper class, but yet another layer of OOP would be ugly here.

Are you suggesting that we pull the following code from `handle_unknown_identity` into helper initialization?
```python
        if not self.user.is_authenticated:
            acting_user = self._get_user(identity)
            login_form = AuthenticationForm(
                # stuff
            )
        else:
            acting_user = self.user
            login_form = None
```
Later there are some state changes to `acting_user` that would need to be tweaked, but I think I like it.